src/tokens.md

```diff
@@ -216,16 +216,51 @@ literal_. The grammar for recognizing the two kinds of literals is mixed.
 
 #### Integer literals
 
+> **<sup>Lexer</sup>**  
+> INTEGER_LITERAL :  
+> &nbsp;&nbsp; ( DEC_LITERAL | BIN_LITERAL | OCT_LITERAL | HEX_LITERAL )
+>              INTEGER_SUFFIX<sup>?</sup>
+>   
+> DEC_LITERAL :  
+> &nbsp;&nbsp; DEC_DIGIT (DEC_DIGIT|`_`)<sup>\*</sup>  
+>  
+> BIN_LITERAL :  
+> &nbsp;&nbsp; `0b` (BIN_DIGIT|`_`)<sup>\*</sup> BIN_DIGIT (BIN_DIGIT|`_`)<sup>\*</sup>  
+>  
+> OCT_LITERAL :  
+> &nbsp;&nbsp; `0o` (OCT_DIGIT|`_`)<sup>\*</sup> OCT_DIGIT (OCT_DIGIT|`_`)<sup>\*</sup>  
+>  
+> HEX_LITERAL :  
+> &nbsp;&nbsp; `0x` (HEX_DIGIT|`_`)<sup>\*</sup> HEX_DIGIT (HEX_DIGIT|`_`)<sup>\*</sup>  
+>  
+> BIN_DIGIT : [`0`-`1` `_`]  
```

Comment:
As you have the `or underscore` in `BIN_LITERAL`, there shouldn't be an underscore in `BIN_DIGIT` or the other digits.